Project21/Project21/intlist.c

```diff
@@ -0,0 +1,212 @@
+   while (current != NULL) {
+      next = current->next;
+      free (current);
+      current = next;
+   }
+   list->head = NULL;
+   list->deleted = true;
+   return 0;
+}
+
+/// <summary>Removes the first occurrence of specified value in list</summary>
+/// <param name="head"></param>
+/// <param name="data"></param>
+/// <returns>Error code (if any)</returns>
+int Remove (LinkedList* list, int data) {
+   // checks if the list is deleted
+   if (list->deleted == true) return E_LIST_DOES_NOT_EXIST;
+   //Checks if the list is empty
+   Node* temp = list->head;
+   Node* pre = NULL;
+   if (temp == NULL) return E_EMPTY_LIST;
+   //Check if the data in head(1st node) is to be removed
+   if (temp != NULL && temp->data == data) {
+      list->head = temp->next;
+      free (temp);
+      return 0;
+   }
+   //if the list is not empty, it iterates to remove the data at spec.index
+   while (temp != NULL && temp->data != data) {//if the list is not empty, it iterates to remove the data at spec. index
+      pre = temp;
+      temp = temp->next;
+   }
+   //checks if index exceeds list count
+   if (temp != NULL) {
+      pre->next = temp->next;
+      free (temp);
+      return 0;
+   }
+   else {
+      return E_VALUE_NOT_FOUND;
+   }
+}
+/// <summary>Removes the value at a specified index</summary>
+/// <param name="head"></param>
+/// <param name="index"></param>
+/// <returns>Error code (if any)</returns>
+int RemoveAt (LinkedList* list, int index) {
+   //checks if the list is deleted
+   if (list->deleted == true) return E_LIST_DOES_NOT_EXIST;
+   //Checks if the list is empty
+   Node* temp = list->head;
+   Node* pre = NULL;
+   if (temp == NULL) return E_EMPTY_LIST;
+   //Check if the data in head(1st node) is to be removed
+   if (index == 0) {
+      list->head = temp->next;
+      free (temp);
+      return 0;
+   }
+   //if the list is not empty, it iterates to remove the data at spec. index
+   for (int i = 0; temp != NULL && i < index; i++) {
+      pre = temp;
+      temp = temp->next;
+   }
+   //checks if index exceeds list count
+   if (temp != NULL && pre != NULL) {
+      pre->next = temp->next;
+      free (temp);
+      return 0;
+   }
+   else {
+      return E_INDEX_OUT_OF_RANGE;
+   }
+}
+
+/// <summary>Inserts the value at a specified index</summary>
+/// <param name="head"></param>
+/// <param name="data"></param>
+/// <param name="index"></param>
+/// <returns>Error code (if any)</returns>
+int Insert (LinkedList* list, int data, int index) {
+   //checks if the list is deleted
+   if (list->deleted == true) return E_LIST_DOES_NOT_EXIST;
+   // Checks if the inserted node is going to be the head
+   Node* newNode = CreateNode (data);
+   if (index == 0) {
+      newNode->next = list->head;
+      list->head = newNode;
+      return 0;
+   }
+   //if the list is not empty, it iterates to get the data at spec. index
+   Node* temp = list->head;
+   for (int i = 0; i < index - 1 && temp != NULL; i++) {

```

Comment:
temp will not be null if you check for index beforehand. You can simply this. Also you do not need to add {} around single line statements